### DISCUSSION TITLE
How to tackle this problem - emulating a organization and its layers of roles?

### DISCUSSION THREAD

**Discussion by charleslcso** [Budibase/budibase#9987] [Help] [answered]
Further to my other question, I need to build an app for a company, which has superadmin (that's me), manager, admin, staff roles. Managers can see all records from a group of staffs, and each staff can only see their own records. Admin can operate on some common settings/parameters.

The managers of each respective team also has a dashboard, of which is a consolidated view/analytical view of the data from each staff. That means, under manager account, there is a dashboard page for each of his staff.

From my limited playtime with Budibase, I believe each staff, manager, admin should has his own Budibase account. Under a staff's account, I should add his manager to the Users list. That should be OK. 

Now if the manager logs into his account, is it possible for him to see the data from each of his staff?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @charleslcso,

Overall everything listed here can be done with Budibase. Its all dependant on how you structure your tables and link everything together.
- Custom roles can easily be created on a project basis, so you could add the manger, staff roles and any additional roles you need. The super admin I would just rename the admin role and assign that to yourself. Although whenever you signup and create your account you are technically already a super admin. https://docs.budibase.com/docs/user-roles#adding-custom-roles (I've supplied a screenshot of a custom role being used within a table below.)
- Like previously stated, depending on how you make the tables, you can really easily get staff related to managers and would be able to see all related information to them.
- You can really easily add different users, you can do this on a user by user basis or bulk add users through a CSV. (Bulk adding would mean you don't have to go through each and assign their roles etc.) Additional you could split this up into multiple import to allow for better grouping of the staff, managers etc.

**Customer user roles project basis**
<img width="286" alt="custom user roles" src="https://user-images.githubusercontent.com/126772285/224654260-756617da-5b50-4628-a956-8f3eefaa7895.png">

**Adding users**
<img width="866" alt="Adding Users" src="https://user-images.githubusercontent.com/126772285/224654402-ef6ba516-d6fc-44b3-98f0-9a7e4b353af5.png">
